scafi-core/src/test/scala/it/unibo/scafi/test/environment/Node.scala

```diff
@@ -2,29 +2,29 @@ package it.unibo.scafi.test.environment
 
 import scala.compiletime.uninitialized
 
-import it.unibo.scafi.context.AggregateContext
 import it.unibo.scafi.message.{ Export, Import }
 import it.unibo.scafi.runtime.ScafiEngine
 import it.unibo.scafi.runtime.network.NetworkManager
 
-class Node[R, Context <: AggregateContext { type DeviceId = Int }](
-    val environment: Environment[R, Context],
+class Node[Result, Context <: IntAggregateContext, Network <: IntNetworkManager](
+    val environment: Environment[Result, Context, Network],
     val id: Int,
     val retain: Int,
-    private val contextFactory: (Int, NetworkManager { type DeviceId = Int }) => Context,
-    private val program: (Context, Environment[R, Context]) ?=> R,
+    private val contextFactory: (Int, Network) => Context,
+    private val program: (Context, Environment[Result, Context, Network]) ?=> Result,
+    private val networkManagerFactory: Node[Result, Context, Network] => Network,
 ):
-  private val nodeNetworkManager = NodeNetworkManager()
-  private var currentResult: R = uninitialized
-  private val engine = ScafiEngine(id, nodeNetworkManager, contextFactory): ctx ?=>
+  lazy val networkManager: Network = networkManagerFactory(this)
```

Comment:
Why `lazy` here?

Reply:
otherwise I get a warning (and error) because I'm passing `this` as an argument but the compiler cannot verify the object is fully initialized.